sys/pool_alloc.c

```diff
@@ -0,0 +1,235 @@
+#include <stdc.h>
+#include <vm.h>
+#include <physmem.h>
+#include <queue.h>
+#include <bitstring.h>
+#include <common.h>
+#include <pool_alloc.h>
+
+#define PI_MAGIC_WORD 0xcafebabe
+
+#define PP_FREED_WORD 0xdeadbeef
+
+#define GET_PI_AT_IDX(slab, i, size)                                           \
+  (pool_item_t *)((unsigned long)slab + (slab->ph_start) +                     \
+                  (i) * (size + sizeof(pool_item_t)))
+
+#define ALIGN_TO_BINARY_POW(x, a) ((((x) >> (a)) + 1) << (a))
```

Comment:
Correct me if I'm wrong, but I think that if `x` is already aligned, then this macro will return a needlessly larger value. Also you may want to use `align` from `common.h`.